packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/500.tsx

```diff
@@ -0,0 +1,46 @@
+import { sitecorePagePropsFactory } from 'lib/page-props-factory';
+import Layout from 'src/Layout';
+import config from 'temp/config';
+import {
+  ComponentPropsContext,
+  GraphQLErrorPagesService,
+  SitecoreContext,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { componentFactory } from 'temp/componentFactory';
+import { SitecorePageProps } from 'lib/page-props';
+import ServerError from 'src/ServerError';
+
+const Custom500 = (props: SitecorePageProps): JSX.Element => {
+  if (!(props && props.layoutData)) {
+    return <ServerError />;
```

Comment:
If it's 500 - you will never have layoutData, I assume and the markup below will not be rendered, did you verify that?

Reply:
I think this is correct. layoutData will be from your configured serverErrorPage (at build time). <ServerError /> is the fallback in the case that there is no serverErrorPage configured.

Reply:
Makes sense to me